### PR TITLE
Photo validation (size and resoulution)

### PR DESCRIPTION
**Pull request by OH72**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java [hunk 1]
```diff
@@ -23,14 +25,37 @@
...
+    private static final Long IMAGE_SIZE_B = IMAGE_SIZE_MB * 1024 * 1024;
+    private static final Long MIN_IMAGE_WIDTH = 200L;
+    private static final Long MIN_IMAGE_HEIGHT = 200L;
 
     @Override
     public String uploadImage(String uploadDir, String fileName, MultipartFile multipartFile) {
         Path uploadPath = Paths.get(uploadDir);
 
+        if(multipartFile.getSize() > IMAGE_SIZE_B){
+            throw new IncorrectInputException(String.format(IMAGE_SIZE_EXCEPTION, IMAGE_SIZE_B, multipartFile.getSize()));
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java:42]: throw exception if size is larger than allowed

File: src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java [hunk 2]
```diff
@@ -23,14 +25,37 @@
...
+
+        try {
+            BufferedImage bufferedImage = ImageIO.read(multipartFile.getInputStream());
+            int width = bufferedImage.getWidth();
+            int height = bufferedImage.getHeight();
+            if(MIN_IMAGE_WIDTH < 200){
+                throw new IncorrectInputException(String.format(IMAGE_RESOLUTION_EXCEPTION, "width", width));
+            }
+            if(MIN_IMAGE_HEIGHT < 200){
+                throw new IncorrectInputException(String.format(IMAGE_RESOLUTION_EXCEPTION, "height", height));
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java:53]: throw exception if height is smaller than allowed

File: src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java [hunk 3]
```diff
@@ -23,14 +25,37 @@
...
+        if(multipartFile.getSize() > IMAGE_SIZE_B){
+            throw new IncorrectInputException(String.format(IMAGE_SIZE_EXCEPTION, IMAGE_SIZE_B, multipartFile.getSize()));
+        }
+
+        try {
+            BufferedImage bufferedImage = ImageIO.read(multipartFile.getInputStream());
+            int width = bufferedImage.getWidth();
+            int height = bufferedImage.getHeight();
+            if(MIN_IMAGE_WIDTH < 200){
+                throw new IncorrectInputException(String.format(IMAGE_RESOLUTION_EXCEPTION, "width", width));
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java:50]: throw exception if widthis smaller than allowed

File: src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java [hunk 4]
```diff
@@ -23,14 +25,37 @@
...
     private final String DIRECTORY_CREATE_EXCEPTION = "Could not create directory with name: %s";
+    private final String IMAGE_SIZE_EXCEPTION = "Max image size should be %d bytes, your image size is %d bytes";
+    private final String IMAGE_RESOLUTION_EXCEPTION = "Image %s should be more than %d";
     private final String UPLOAD_LOCATION = "/upload";
     private static final String UPLOAD_PLUG = "/upload/test/test.png";
-
+    private static final Long IMAGE_SIZE_MB = 5L;
+    private static final Long IMAGE_SIZE_B = IMAGE_SIZE_MB * 1024 * 1024;
+    private static final Long MIN_IMAGE_WIDTH = 200L;
+    private static final Long MIN_IMAGE_HEIGHT = 200L;
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java:35]: allowed constants

File: src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java [hunk 5]
```diff
@@ -23,14 +25,37 @@
 public class FileUploadServiceImpl implements FileUploadService {
     private final String FILE_UPLOAD_EXCEPTION = "Could not save image file: %s";
     private final String DIRECTORY_CREATE_EXCEPTION = "Could not create directory with name: %s";
+    private final String IMAGE_SIZE_EXCEPTION = "Max image size should be %d bytes, your image size is %d bytes";
+    private final String IMAGE_RESOLUTION_EXCEPTION = "Image %s should be more than %d";
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java:29]: exception messages

**Review by VictoriaAmazonka — CHANGES_REQUESTED** [review]
e.printStackTrace(); -we have to use logger to log messages.

**Comment by OH72** [comment]
> 
> 
> e.printStackTrace(); -we have to use logger to log messages.

I replaced it with log.error.

**Review by VictoriaAmazonka — APPROVED** [review]

**Pull request merged**